tests/DependencyInjection/Compiler/BusBuildersFromConfig.php

```diff
@@ -0,0 +1,134 @@
+<?php
+declare(strict_types=1);
+
+namespace League\Tactician\Bundle\Tests\DependencyInjection\Compiler\BusBuilder;
+
+use PHPUnit\Framework\TestCase;
+use League\Tactician\Bundle\DependencyInjection\Compiler\BusBuilder\BusBuilder;
+use League\Tactician\Bundle\DependencyInjection\Compiler\BusBuilder\BusBuildersFromConfig;
+
+final class BusBuildersFromConfigTest extends TestCase
+{
+    public function test config leads to builder with default for each commandbus()
+    {
+        $builders = BusBuildersFromConfig::convert([
+            'commandbus' => [
+                'default' => [
+                    'middleware' => [
+                        'my.middleware',
+                    ],
+                ],
+                'other' => [
+                    'middleware' => [
+                        'my.other.middleware',
+                    ],
+                ],
+            ],
+        ]);
+
+        $this->assertEquals(
+            new BusBuilder('default', BusBuildersFromConfig::DEFAULT_METHOD_INFLECTOR, ['my.middleware']),
+            $builders->getIterator()['default']
+        );
+        $this->assertEquals(
+            new BusBuilder('other', BusBuildersFromConfig::DEFAULT_METHOD_INFLECTOR, ['my.other.middleware']),
+            $builders->getIterator()['other']
+        );
+    }
+
+    public function test default method inflector can be overrided()
+    {
+        $builders = BusBuildersFromConfig::convert([
+            'method_inflector' => 'other.inflector',
+            'commandbus' => [
+                'default' => [
+                    'middleware' => [
+                        'my.middleware',
+                    ],
+                ],
+                'other' => [
+                    'middleware' => [
+                        'my.other.middleware',
+                    ],
+                ],
+            ],
+        ]);
+
+        $this->assertEquals(
+            new BusBuilder('default', 'other.inflector', ['my.middleware']),
+            $builders->getIterator()['default']
+        );
+    }
+
+    public function test method inflector of each bus can be overrided()
```

Comment:
First time I've ever seen this in the wild. I'm....not sure about this. 😑

Reply:
oh did not realize, I did it mechanically ! My bad, i will update with `_` :+1: